src/main/java/org/wso2/securevault/secret/SecretRepository.java

```diff
@@ -25,6 +25,16 @@
  * Implementation can be any type - file,jdbc
  */
 public interface SecretRepository {
+    /**
+     * Provide the type of the secret repository
+     *
+     * @return secret repository type
+     */
+    default String getType() {
+
+        return "secret repository type";
```

Comment:
can we have a different default type

Reply:
Removed the getType method and instead get the type via the configuration file